lib/rspec/support/output_formatter.rb

```diff
@@ -0,0 +1,46 @@
+module RSpec
+  module Support
+    # Provide additional output details beyond what `inspect` provides when
+    # printing Time, DateTime, or BigDecimal
+    module OutputFormatter
```

Comment:
I think `OutputFormatter` is perhaps not the best name. Within the RSpec ecosystem, "formatter" has a pretty specific meaning of an object that implements rspec-core's formatter protocol and receives notification of events in order to format output.  This isn't like that at all.

Maybe `ObjectDescriber` or `ObjectInspector` is better?  The latter seems particularly appropriate since this is essentially a replacement for a less useful `inspect` call.